network/src/main/java/javasabr/mqtt/network/message/in/SubscribeAckMqttInMessage.java

```diff
@@ -90,7 +104,12 @@ protected Set<MqttMessageProperty> availableProperties() {
   @Override
   protected void applyProperty(MqttMessageProperty property, String value) {
     switch (property) {
-      case REASON_STRING -> reason = value;
+      case REASON_STRING -> {
+        if (reason != null) {
+          alreadyPresentedProperty(property);
```

Comment:
Could we consider to extract common part of `applyProperty` method to parent class in order to reuse it